lib/cachex/warmer.ex

```diff
@@ -67,19 +67,37 @@ defmodule Cachex.Warmer do
       # enforce the behaviour
       @behaviour Cachex.Warmer
 
+      def warm(pid, opts) do
+        if opts[:respect_async_warmers] do
+          GenServer.call(pid, :cachex_warmer, :infinity)
+        else
+          send(pid, :cachex_warmer)
+        end
+      end
```

Comment:
We need to be able to block for `async: false` workers, so I added this api - if you would prefer we could move this if/else block to the `Actions.Warm` module to avoid the additional function here.